### PR TITLE
Mrcjkb/issue#536

### LINKED ISSUE #536: java.lang.NumberFormatException: Character N is neither a decimal digit number after upgrading to 3.6.6
After upgrading from 3.6.5 to 3.6.6 version the following test starts to fail:
```
    @Test
    void test() throws Exception {
        XYChart chart = new XYChart(800, 600);

        List<? extends Number> values = List.of(NaN, NaN, NaN);
        List<? extends Number> times = List.of(1L, 2L, 3L);
        XYSeries series = chart.addSeries("Series", times, values);
        series.setMarker(SeriesMarkers.NONE);

        byte[] bytes = BitmapEncoder.getBitmapBytes(chart, BitmapEncoder.BitmapFormat.PNG);
    }
```

Here is the stacktrace:
```
java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.

	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:518)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:401)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:834)
	at java.base/java.math.BigDecimal.valueOf(BigDecimal.java:1304)
	at org.knowm.xchart.internal.chartpart.AxisTickCalculator_.calculate(AxisTickCalculator_.java:295)
	at org.knowm.xchart.internal.chartpart.AxisTickCalculator_Number.<init>(AxisTickCalculator_Number.java:47)
	at org.knowm.xchart.internal.chartpart.Axis.getAxisTickCalculator(Axis.java:572)
	at org.knowm.xchart.internal.chartpart.Axis.getYAxisWidthHint(Axis.java:349)
	at org.knowm.xchart.internal.chartpart.Axis.preparePaint(Axis.java:159)
	at org.knowm.xchart.internal.chartpart.AxisPair.paint(AxisPair.java:124)
	at org.knowm.xchart.XYChart.paint(XYChart.java:417)
	at org.knowm.xchart.BitmapEncoder.getBufferedImage(BitmapEncoder.java:281)
	at org.knowm.xchart.BitmapEncoder.getBitmapBytes(BitmapEncoder.java:264)
	at yyy.MetricChartCreatorTest.test(MetricChartCreatorTest.java:69)
```

### PR DESCRIPTION
**Pull request by MrcJkb**
Fixes #536

### CODE REVIEW

**Comment by timmolter** [comment]
So I tracked this down and found that you changed the way that the min and max values are overridden in `AxisTickCalculator_` via:

```java
    this.minValue =
        getAxisMinValue(
            styler,
            axisDirection,
            this.axisValues.stream()
                .filter(Objects::nonNull)
                .mapToDouble(x -> x)
                .min()
                .orElseThrow(NoSuchElementException::new));
```
whereas before it was:

```java
    this.minValue =
        getAxisMinValue(
            styler,
            axisDirection,
            minValue);
```
Is there a reason or necessity for your modification?

File: xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_.java [hunk 1]
```diff
@@ -74,30 +74,28 @@
...
                 .mapToDouble(x -> x)
                 .min()
-                .orElseThrow(NoSuchElementException::new));
+                .orElse(minValue));
 
     this.maxValue =
         getAxisMaxValue(
             styler,
             axisDirection,
                 this.axisValues.stream()
```
Comment (timmolter) [inline xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_.java:93]: Just curious, what's the advantage of 

```
this.axisValues.stream()
                .filter(Objects::nonNull)	                .filter(Objects::nonNull)
                .filter(x -> x >= maxValue)
                .mapToDouble(x -> x)	                .mapToDouble(x -> x)
                .max()	                .max()
                .orElseThrow(NoSuchElementException::new));	                .orElse(maxValue)
```

over the original `maxValue` as I had it before?

**Comment by MrcJkb** [comment]
T

> So I tracked this down and found that you changed the way that the min and max values are overridden in `AxisTickCalculator_` via:
> 
> ```java
>     this.minValue =
>         getAxisMinValue(
>             styler,
>             axisDirection,
>             this.axisValues.stream()
>                 .filter(Objects::nonNull)
>                 .mapToDouble(x -> x)
>                 .min()
>                 .orElseThrow(NoSuchElementException::new));
> ```
> 
> whereas before it was:
> 
> ```java
>     this.minValue =
>         getAxisMinValue(
>             styler,
>             axisDirection,
>             minValue);
> ```
> 
> Is there a reason or necessity for your modification?

You are right, that is another bug.
Your excerpt of my change is incomplete. The complete change is as follows:
```
Set<Double> axisValuesWithMinMax = new LinkedHashSet<>();
axisValuesWithMinMax.add(minValue);
axisValuesWithMinMax.addAll(axisValues);
axisValuesWithMinMax.add(maxValue);
this.axisValues = new ArrayList<>(axisValuesWithMinMax);
    this.minValue =
    getAxisMinValue(
        styler,
        axisDirection,
        this.axisValues.stream()
            .filter(Objects::nonNull)
            .mapToDouble(x -> x)
            .min()
            .orElseThrow(NoSuchElementException::new));
```

The stream for determining the min and max values is a remnant from before https://github.com/knowm/XChart/pull/531/, in which the constructor taking only the `axisValues` List determined the min and max values from the `axisValues`, ignoring the overridden min and max values.
The `axisValues` List is used to determine the tick locations in case the values are all equally spaced.

My fix for issue #530 only took into account cases in which overridden min and max values are greater than, and smaller than the `avisValues`, respectively. Now that the overridden `minValue` and `maxValue` are passed to the constructor, it is no longer necessary to determine them from the `axisValues`.

I have adjusted my PR accordingly.

File: xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_.java (hunk 1 above)
Comment (MrcJkb) [reply xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_.java]: This was just a test that I pushed too early.

**Comment by timmolter** [comment]
Alles klar. Thanks!

**Pull request merged**